### PR TITLE
Add article for Control Center Development Mode

### PR DESCRIPTION
**Pull request by heruan**
This adds documentation on how to run a Vaadin app in development mode with Control Center.

Closes https://github.com/vaadin/control-center/issues/459

### CODE REVIEW

File: articles/control-center/dev-mode/index.adoc [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+If you encounter issues while setting up or running Telepresence, refer to the https://www.telepresence.io/docs/latest/troubleshooting/[Telepresence Troubleshooting Guide] for solutions to common problems.
+
+For macOS users, especially on the latest versions, additional network and security configurations may be necessary. See the https://www.telepresence.io/docs/latest/troubleshooting/#macos[macOS-specific troubleshooting section] for guidance.
+
+== Additional Resources
+
+* https://www.telepresence.io/docs/latest/[Telepresence Documentation]
+* https://www.vaadin.com/docs/[Vaadin Documentation]
+* https://www.telepresence.io/docs/latest/howtos/intercepts/[Understanding Intercepts in Telepresence]
+* https://www.telepresence.io/docs/latest/howtos/ports/[Managing Ports with Telepresence]
```
Comment (MatthewVaadin) [inline articles/control-center/dev-mode/index.adoc:161]: URL leads to a 404 page. I'm not sure what would replace this.

**Review by MatthewVaadin — CHANGES_REQUESTED** [review]
Need to add "Telepresence" to the dictionary.

File: articles/control-center/dev-mode/index.adoc [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+=== Install Telepresence in the Cluster
+
+Install the Telepresence Traffic Manager in your cluster using Helm:
+
+[source,bash]
+----
+telepresence helm install
+----
+
+For detailed installation options, refer to the https://www.telepresence.io/docs/latest/install/cluster/[Telepresence Cluster Installation documentation].
```
[the remark this answers is not in the document]
Comment (heruan) [reply articles/control-center/dev-mode/index.adoc]: Fixed.

File: articles/control-center/dev-mode/index.adoc [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+== Preparing Your Application
+
+To ensure your application can access cluster resources while running locally, update your application's configuration.
+
+Create or modify the `application.yaml` file in `src/main/resources/` with the following content:
+
+[source,yaml,subs="+quotes,verbatim"]
```
[the remark this answers is not in the document]
Comment (heruan) [reply articles/control-center/dev-mode/index.adoc:71]: What do you mean adding a version?

File: articles/control-center/dev-mode/index.adoc [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    name: 'my-vaadin-app'
+  cloud:
+    kubernetes:
+      client:
+        namespace: 'control-center'
+  main:
+    cloud-platform: 'KUBERNETES'
+----
+
+Replace `'my-vaadin-app'` with the actual name of your application and `'control-center'` with the namespace where Control Center is deployed.
```
[the remark this answers is not in the document]
Comment (heruan) [reply articles/control-center/dev-mode/index.adoc:84]: We already have some admonitions, I don't want to have too much. Using `my-vaadin-app` should be clear enough to be an example value.

File: articles/control-center/dev-mode/index.adoc (hunk 1 above)
Comment (heruan) [reply articles/control-center/dev-mode/index.adoc]: Fixed all the broken links.

File: articles/control-center/dev-mode/index.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (MatthewVaadin) [reply articles/control-center/dev-mode/index.adoc:71]: Like we do in (for example) Kubernetes Kit documentation:
```
[.example]
--
.application.properties
[source,properties]
----
vaadin.kubernetes.hazelcast.service-name=hazelcast-service
vaadin.kubernetes.hazelcast.namespace=namespace
----

.application.yaml
[source,yaml]
----
vaadin:
  kubernetes:
    hazelcast:
      service-name: hazelcast-service
      namespace: namespace
----
--
```

[the remark this answers is not in the document]
Comment (heruan) [reply articles/control-center/dev-mode/index.adoc:86]: Oh right, I thought you meant adding a version like `vaadin.version=1.2.3` 😅 Done!

**Review by MatthewVaadin — APPROVED** [review]

**Pull request merged**